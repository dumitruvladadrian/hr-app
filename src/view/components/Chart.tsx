import React from 'react';
import {
	CartesianGrid,
	Legend,
	Line,
	LineChart,
	ResponsiveContainer,
	Tooltip,
	XAxis,
	YAxis,
} from 'recharts';

interface ChartDataPoint {
	name: string;
	average: number;
}

interface ChartProps {
	computeData: () => Array<ChartDataPoint>;
}

// basic pre-configured chart component
export const Chart = ({ computeData }: ChartProps) => (
	<div>
		<div style={{ width: 1500, height: '80vh' }}>
			<ResponsiveContainer width="100%" height="100%">
				<LineChart
					width={500}
					height={300}
					data={computeData()}
					margin={{
						top: 5,
						right: 30,
						left: 20,
						bottom: 5,
					}}
				>
					<CartesianGrid strokeDasharray="3 3" />
					<XAxis dataKey="name" angle={-70} textAnchor="end" height={500} interval={0} />
					<YAxis dataKey="average" />
					<Tooltip />
					<Legend />
					<Line type="monotone" dataKey="average" stroke="#8884d8" activeDot={{ r: 8 }} />
				</LineChart>
			</ResponsiveContainer>
		</div>
	</div>
);
