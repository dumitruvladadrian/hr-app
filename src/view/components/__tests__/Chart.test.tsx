import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import React from 'react';
import { Chart } from '../Chart';

test('Checking App snapshot', () => {
	const computedData = () => [
		{ average: 30, name: 'Electrical Products' },
		{
			average: 43,
			name: 'Telecommunications Equipment',
		},
	];
	const chart = render(<Chart computeData={computedData} />);
	expect(chart.container.firstChild).toMatchSnapshot();
});
