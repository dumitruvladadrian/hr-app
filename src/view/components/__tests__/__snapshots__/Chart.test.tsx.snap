// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking App snapshot 1`] = `
<div>
  <div
    style="width: 1500px; height: 80vh;"
  >
    <div
      class="recharts-responsive-container"
      style="width: 100%; height: 100%;"
    />
  </div>
</div>
`;
